cpp/src/dictionary/add_keys.cu

```diff
@@ -58,25 +59,28 @@ std::unique_ptr<column> add_keys(
   auto combined_keys =
     cudf::detail::concatenate(std::vector<column_view>{old_keys, new_keys}, stream);
   // sort and remove any duplicates from the combined keys
-  // drop_duplicates([a,b,c,d,f,d,b,e]) = [a,b,c,d,e,f]
-  auto table_keys = cudf::detail::drop_duplicates(table_view{{combined_keys->view()}},
-                                                  std::vector<size_type>{0},  // only one key column
-                                                  duplicate_keep_option::KEEP_FIRST,
-                                                  null_equality::EQUAL,
-                                                  null_order::BEFORE,
-                                                  stream,
-                                                  mr)
-                      ->release();
-  std::unique_ptr<column> keys_column(std::move(table_keys.front()));
+  // unordered_drop_duplicates([a,b,c,d,f,d,b,e]) = [a,b,c,d,e,f]
```

Comment:
This comment implies that the result of `unordered_drop_duplicates` is sorted, but the sorting happens in a second step. Maybe write:
```
  // Drop duplicates from the combined keys, then sort the result.
  // sort(unordered_drop_duplicates([a,b,c,d,f,d,b,e])) = [a,b,c,d,e,f]
```